### PR TITLE
Close Keepass before upgrading or uninstalling KeePass

### PR DESCRIPTION
**Pull request by johannes2704**
Like discussed at https://chocolatey.org/packages/keepass.install

I have created the ChocolateyBeforeModify.ps1 script to automatically close KeePass before upgrading or uninstalling the package.

### CODE REVIEW

**Comment by pauby** [comment]
@johannes2704 Thanks for submitting this.

What happens if the user has an unsaved entry in Keepass when you force it to close?

**Comment by johannes2704** [comment]
With the first commit the entry will be lost.
I have changed the method now to the official keepass command line option.

https://keepass.info/%0D/help/base/cmdline.html
If a user has now an unsaved entry the user will be asked to save the database or not.

**Comment by pauby** [comment]
That looks good. Added a comment on the code. 

My concern is that Choco just sits there waiting for KeePass asking the user to save the database ... and then times out and fails.

Without testing this I'd assume that if KeePass is open and we attenpt to close it it will jump to the foreground asking the user to save that unsaved entry so the user sees it and knows what to do? If not can we add some text displayed on the console saying that it's going to try and close KeePass and that the user must do something?

Other than that, excellent job. Thanks for working on this.

**Comment by johannes2704** [comment]
I have added a message to inform the user.

But the script does not wait until KeePass is closed. If KeePass is open during installation, the installation will fail. Just as it was.

**Comment by pauby** [comment]
@johannes2704 Can you have a look the other comments on this PR?

**Comment by johannes2704** [comment]
which one? I can only see your comments.

File: automatic/keepass.install/tools/chocolateybeforemodify.ps1 [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿# Automatically close KeePass if KeePass is running. If the database has unsaved changes the user will be prompted to save the database.
+# Works only with the default installation paths
+if (Get-Process -Name "Keepass" -ErrorAction "SilentlyContinue" ) { 
+    Write-Output "KeePass is running. Try to close KeePass automatically. If you have unsaved entries, you will be prompted to save them."
+    if (Test-Path "${env:ProgramFiles(x86)}\KeePass Password Safe 2\KeePass.exe") {
+        & "${env:ProgramFiles(x86)}\KeePass Password Safe 2\KeePass.exe" "--exit-all"
+    } else {
```
Comment (pauby) [inline automatic/keepass.install/tools/chocolateybeforemodify.ps1]: Can we just make this an elseif rather than an else then an if?

**Comment by pauby** [comment]
> which one? I can only see your comments.

Can you see it now?

**Comment by johannes2704** [comment]
Now I can see it. I have changed the code to an elseif statement.

File: automatic/keepass.install/tools/chocolateybeforemodify.ps1 (hunk 1 above)
Comment (johannes2704) [reply automatic/keepass.install/tools/chocolateybeforemodify.ps1]: Done.

**Pull request merged**